examples/chef/common/chef-rvc-mode-delegate.cpp

```diff
@@ -106,17 +108,61 @@ void RvcRunMode::Shutdown()
     }
 }
 
+chip::Protocols::InteractionModel::Status chefRvcRunModeWriteCallback(chip::EndpointId endpointId, chip::ClusterId clusterId,
+                                                                      const EmberAfAttributeMetadata * attributeMetadata,
+                                                                      uint8_t * buffer)
+{
+    chip::Protocols::InteractionModel::Status ret = Protocols::InteractionModel::Status::Success;
+
+    VerifyOrDie(endpointId == 1); // this cluster is only enabled for endpoint 1.
+    VerifyOrDie(gRvcRunModeInstance != nullptr);
+
+    ModeBase::Instance * clusterInstance = gRvcRunModeInstance;
+    chip::AttributeId attributeId        = attributeMetadata->attributeId;
+
+    switch (attributeId)
+    {
+    case chip::app::Clusters::RvcRunMode::Attributes::CurrentMode::Id: {
+        uint8_t m                                        = static_cast<uint8_t>(buffer[0]);
+        chip::Protocols::InteractionModel::Status status = clusterInstance->UpdateCurrentMode(m);
+        if (chip::Protocols::InteractionModel::Status::Success == status)
+        {
+            break;
+        }
+        ret = chip::Protocols::InteractionModel::Status::UnsupportedWrite;
+        ChipLogError(DeviceLayer, "Invalid Attribute Update status: %d", static_cast<int>(status));
+    }
+    break;
+    default:
+        ret = chip::Protocols::InteractionModel::Status::UnsupportedAttribute;
+        ChipLogError(DeviceLayer, "Unsupported Attribute ID: %d", static_cast<int>(attributeId));
+        break;
+    }
+
+    return ret;
+}
+
+chip::Protocols::InteractionModel::Status chefRvcRunModeReadCallback(chip::EndpointId endpointId, chip::ClusterId clusterId,
+                                                                     const EmberAfAttributeMetadata * attributeMetadata,
+                                                                     uint8_t * buffer, uint16_t maxReadLength)
+{
+
+    return chip::Protocols::InteractionModel::Status::Success;
```

Comment:
is this guaranteed to be always correct? We do not alter `buffer` at all. Should we at least memset to 0? A comment should explain why a read that is a noop is to be considered a success.

Reply:
I'm not sure whether it's neccessary since it's a call back register which does nothing.

Reply:
The "does nothing" is the bit that is unclear to me. The way I understand it is that we ask "please fill buffer with the data" and we return success, basically saying "success".

It seems odd that we say success but we leave buffer in an unfilled/undefined state. Would a `memset(buffer, 0, maxReadLength)` be correct? Should we fill buffer with a specific run more instead?

Just leaving it to uninitialized (I imagine callers will just allocate stack space for this) seems wrong and non-deterministic.